kevm-pyk/src/kevm_pyk/solc_to_k.py

```diff
@@ -132,8 +141,10 @@ def _get_method_abi(_mname: str) -> Dict:
         _methods = []
         for msig in method_ids:
             mname = msig.split('(')[0]
+            margs_cs = msig.split('(')[1][:-1]
+            margs = [] if margs_cs == '' else margs_cs.split(',')
             mid = int(method_ids[msig], 16)
-            _m = Contract.Method(mname, mid, _get_method_abi(mname), contract_name, self.sort_method)
+            _m = Contract.Method(mname, mid, _get_method_abi(mname, margs), contract_name, self.sort_method)
             _methods.append(_m)
         self.methods = tuple(_methods)
```

Comment:
`_get_method_abi` is only used for this purpose, so I would suggest the following change:

- Just pass in `msig` directly to `Contract.Method.__init__` as teh third argument.
- Move the processing of the signature into the `Contract.Method.__init__` constructor directly. Either by moving the method `_get_method_abi` into the `Contract.Method` class and calling it, or just by inlining its logic directly into the `Method` constructor.

Reply:
I moved the signature processing and method lookup logic into the Method  constructor.